src/gui/folderman.cpp

```diff
@@ -720,22 +720,41 @@ static QString canonicalPath(const QString &path)
     return selFile.canonicalFilePath();
 }
 
-static QString checkPathForSyncRootMarkingRecursive(const QString &path)
-{
-    auto existingTag = Utility::getDirectorySyncRootMarking(path);
-    if (!existingTag.isEmpty()) {
-        return FolderMan::tr("Folder '%1' is already in use by application %2!").arg(path, existingTag);
+static QString checkPathForSyncRootMarkingRecursive(const QString &path, FolderMan::NewFolderType folderType, const QUuid &accountUuid)
+{
+    std::pair<QString, QUuid> existingTags = Utility::getDirectorySyncRootMarkings(path);
```

Comment:
Why do we need to differentiate for FolderMan::NewFolderType?

Reply:
We need to know if it's a Spaces folder we're checking for: that one is allowed (preferred!) to end up in a folder that's tagged as a sync root for Spaces.

Reply:
done